From 934e59d57a5172a4b98d5cdbae9e3966608b1846 Mon Sep 17 00:00:00 2001
From: Wang ShaoBo <bobo.shaobowang@huawei.com>
Date: Fri, 30 Jul 2021 15:44:22 +0800
Subject: [PATCH] Revert "kretprobe: check re-registration of the same
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 934e59d57a5172a4b98d5cdbae9e3966608b1846

 kretprobe earlier"

hulk inclusion
category: bugfix
bugzilla: 31369

-----------------------------------------------

This reverts commit ec02feff173db21bc769d0f4f3f2f011616cca3c.
Following patche mainline has fixed this issue:
0188b87899ff ("kretprobe: Avoid re-registration of the same kretprobe earlier")

Signed-off-by: Wang ShaoBo <bobo.shaobowang@huawei.com>
Reviewed-by: Jian Cheng <cj.chengjian@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 kernel/kprobes.c | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/kernel/kprobes.c b/kernel/kprobes.c
index fb63533f62ff..f590e9ff3706 100644
--- a/kernel/kprobes.c
+++ b/kernel/kprobes.c
@@ -2137,11 +2137,6 @@ int register_kretprobe(struct kretprobe *rp)
 		}
 	}
 
-	/* Return error if it's being re-registered. */
-	ret = check_kprobe_rereg(&rp->kp);
-	if (ret)
-		return ret;
-
 	rp->kp.pre_handler = pre_handler_kretprobe;
 	rp->kp.post_handler = NULL;
 	rp->kp.fault_handler = NULL;
-- 
2.26.2

